pkg/handler/enqueue_annotation.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package handler
+
+import (
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var _ EventHandler = &EnqueueRequestForAnnotation{}
+
+const (
+	// NamespacedNameAnnotation defines the annotation that will be used to get the primary resource namespaced name.
+	// The handler will use this value to build the types.NamespacedName object used to enqueue a Request when an event
+	// to update, to create or to delete the observed object is raised. Note that if only one value be informed without
+	// the "/"  then it will be set as the name of the primary resource.
+	NamespacedNameAnnotation = "sigs.k8s.io/primary-resource-namespace"
```

Comment:
> sigs.k8s.io/primary-resource-namespace

Probably better to use `watch.kubebuilder.io/owner-namespaced-name` or something similar.
Reasons: 1) using `kubebuilder.io` to better show the owner of this annotation. 2) It's misleading to call a `namespacedName` `namespace`.

Reply:
solved